src/types.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Common configuration that will be obtained from
+ * environment variables if not explicitly provided.
+ */
+export interface OAuthEnvConfig {
+  /** The client ID provided by the authorization server. */
+  clientId: string;
+
+  /** The client secret provided by the authorization server, if using a confidential client. */
+  clientSecret: string;
+
+  /** The custom domain of the authorization server, if supported by the provider. */
+  domain?: string;
+}
+
+/**
+ * Information supplied by a provider but not directly necessary for
+ * the OAuth2 process.
+ */
+export interface OAuthProviderInfo {
+  /** The name of the provider (single PascalCase name that matches the function name and uppercase env var prefix) */
+  name: string;
```

Comment:
I've added this addition info for two purposes:
* The env var prefix is derived from this (uppercase).
* The user can support multiple providers in their app, and switch UI features or API calls based on this value without having to keep a separate var of the provider chosen, the demo actually demonstrates this nicely.